ui/components/app/fee-details-component/fee-details-component.js

```diff
@@ -0,0 +1,186 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  IconColor,
+  JustifyContent,
+  Size,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  BUTTON_VARIANT,
+  Box,
+  Button,
+  IconName,
+  Text,
+} from '../../component-library';
+import TransactionDetailItem from '../transaction-detail-item/transaction-detail-item.component';
+import { getIsMultiLayerFeeNetwork, getPreferences } from '../../../selectors';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import LoadingHeartBeat from '../../ui/loading-heartbeat';
+import UserPreferencedCurrencyDisplay from '../user-preferenced-currency-display/user-preferenced-currency-display.component';
+import { PRIMARY, SECONDARY } from '../../../helpers/constants/common';
+import { addHexes } from '../../../../shared/modules/conversion.utils';
+import fetchEstimatedL1Fee from '../../../helpers/utils/optimism/fetchEstimatedL1Fee';
+import { useGasFeeContext } from '../../../contexts/gasFee';
+
+export default function FeeDetailsComponent({
+  txData,
+  supportsEIP1559,
+  useCurrencyRateCheck,
+  shouldShow = true,
+}) {
+  const [expandFeeDetails, setExpandFeeDetails] = useState(false);
+  const [estimatedL1Fees, setEstimatedL1Fees] = useState(null);
+
+  const isMultiLayerFeeNetwork = useSelector(getIsMultiLayerFeeNetwork);
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+
+  const t = useI18nContext();
+
+  const {
+    maximumCostInHexWei: hexMaximumTransactionFee,
+    minimumCostInHexWei: hexMinimumTransactionFee,
+  } = useGasFeeContext();
+  useEffect(() => {
+    if (isMultiLayerFeeNetwork) {
+      fetchEstimatedL1Fee(txData?.chainId, txData)
+        .then((result) => {
+          setEstimatedL1Fees(result);
+        })
+        .catch((_err) => {
+          setEstimatedL1Fees(null);
+        });
+    }
+  }, [isMultiLayerFeeNetwork, txData]);
+
+  const getTransactionFeeTotal = useMemo(() => {
+    if (isMultiLayerFeeNetwork) {
+      return addHexes(hexMaximumTransactionFee, estimatedL1Fees || 0);
+    }
+
+    return hexMaximumTransactionFee;
+  }, [isMultiLayerFeeNetwork, hexMaximumTransactionFee, estimatedL1Fees]);
+
+  const renderTotalDetailText = useCallback(
+    (value) => {
+      return (
+        <div className="confirm-page-container-content__total-value">
+          <LoadingHeartBeat estimateUsed={txData?.userFeeLevel} />
+          <UserPreferencedCurrencyDisplay
+            type={SECONDARY}
+            key="total-detail-text"
+            value={value}
+            hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
+          />
+        </div>
+      );
+    },
+    [txData, useNativeCurrencyAsPrimaryCurrency],
+  );
+
+  const renderTotalDetailTotal = useCallback(
```

Comment:
Can we get a better name for this? It is hard to grasp what `TotalDetailTotal` means..

Reply:
sure